src/django/api/views/export.py

```diff
@@ -0,0 +1,144 @@
+
+    return response
+
+
+@api_view(["GET"])
+@permission_classes([IsAuthenticated, UserIsAdmin])
+def boundaries_all(request):
+    query = """
+    WITH features AS (
+        SELECT json_build_object(
+            'type',       'Feature',
+            'id',         b.id,
+            'geometry',   ST_AsGeoJSON(shape)::json,
+            'properties', json_build_object(
+                'name',            u.name,
+                'pwsid',           u.pwsid,
+                'city',            u.address_city,
+                'state',           u.state_id,
+                'drafted_at',      s.created_at,
+                'drafted_by',      du.email,
+                'submitted_at',    s.submitted_at,
+                'submitted_by',    su.email,
+                'submitted_notes', s.notes,
+                'reviewed_at',     r.reviewed_at,
+                'reviewed_by',     ru.email,
+                'reviewed_notes',  r.notes,
+                'approved_at',     a.approved_at,
+                'approved_by',     au.email,
+                'status',          CASE
+                                       WHEN s.submitted_at IS NULL THEN 'Draft'
+                                       WHEN r.id IS NOT NULL
+                                           AND r.reviewed_at IS NULL
+                                           THEN 'In Review'
+                                       WHEN r.id IS NOT NULL
+                                           AND r.reviewed_at IS NOT NULL
+                                           THEN 'Needs Revisions'
+                                       WHEN a.approved_at IS NOT NULL
+                                           AND a.unapproved_at IS NULL
+                                           THEN 'Approved'
+                                       ELSE 'Submitted'
+                                   END
+            )
+        ) AS feature
+        FROM api_boundary b
+            INNER JOIN api_utility u ON u.id = b.utility_id
+            LEFT JOIN (
+                SELECT MAX(id) AS id, boundary_id
+                FROM api_submission
+                GROUP BY boundary_id) ls ON b.id = ls.boundary_id
+            LEFT JOIN api_submission s ON s.id = ls.id
+                LEFT JOIN api_user du ON s.created_by_id = du.id
+            LEFT JOIN api_user su ON s.submitted_by_id = su.id
+            LEFT JOIN (
+                SELECT MAX(id) AS id, submission_id
+                FROM api_review
+                GROUP BY submission_id) lr ON s.id = lr.submission_id
+            LEFT JOIN api_review r ON r.id = lr.id
+            LEFT JOIN api_user ru ON r.reviewed_by_id = ru.id
+            LEFT JOIN (
+                SELECT MAX(id) AS id, submission_id
+                FROM api_approval
+                GROUP BY submission_id) la ON s.id = la.submission_id
+            LEFT JOIN api_approval a ON a.id = la.id
+            LEFT JOIN api_user au ON a.approved_by_id = au.id
```

Comment:
`LATERAL` subqueries can reference columns from earlier in the `FROM` clause. This lets us use the date to find the latest submission/review rather than the id where the order might not be guaranteed.

More info can be found in `7.2.1.5. LATERAL Subqueries` of [7.2.1. The FROM Clause](https://www.postgresql.org/docs/12/queries-table-expressions.html#QUERIES-FROM).
 
```suggestion
            INNER JOIN api_utility u ON u.id = b.utility_id
            LEFT JOIN LATERAL (
                SELECT * FROM api_submission
                WHERE boundary_id = b.id
                ORDER BY created_at DESC
                LIMIT 1
            ) s ON TRUE
            LEFT JOIN api_user du ON s.created_by_id = du.id
            LEFT JOIN api_user su ON s.submitted_by_id = su.id
            LEFT JOIN LATERAL (
                SELECT * FROM api_review
                WHERE submission_id = s.id
                ORDER BY created_at DESC
                LIMIT 1
            ) r ON TRUE
            LEFT JOIN api_user ru ON r.reviewed_by_id = ru.id
            LEFT JOIN LATERAL (
                SELECT * FROM api_approval
                WHERE submission_id = s.id
                ORDER BY approved_at DESC
                LIMIT 1
            ) a ON TRUE
            LEFT JOIN api_user au ON a.approved_by_id = au.id
```

Reply:
Ah interesting! I hadn't heard of this before. Still don't quite have an intuition after reading up on it. I think of subqueries are row-level operations, and joins as table-level operations, so it's strange to have a subquery in a join statement. I timed them in psql though and the LATERAL JOIN appears to be running faster (???) which does not intuit well.

Nevertheless, the results are solid, and the reasoning makes sense, so I'm going with it.

[ed267d5](https://github.com/azavea/iow-boundary-tool/pull/218/commits/ed267d581e48ea8ff465e25a6fba7f5096d4dca8)